extra/query_propagation.rst

```diff
@@ -0,0 +1,123 @@
+Controlling Query Propagation
+#############################
+
+When the user issues a query, the Citus coordinator partitions it into smaller query fragments where each query fragment can be run independently on a worker shard. This allows Citus to distribute each query across the cluster.
+
+However the way queries are partitioned into fragments (and which queries are propagated at all) varies by the type of query. In some advanced situations it is useful to manually control this behavior. Citus provides utility functions to propagate SQL to workers, shards, or placements.
+
+Running on all Workers
+----------------------
+
+The least granular level of execution is broadcasting a statement for execution on all workers. This is useful for viewing a properties of entire worker databases or creating UDFs uniformly throughout the cluster. For example:
+
+.. code-block:: postgresql
+
+  -- Make a UDF available on all workers
+  SELECT run_command_on_workers($cmd$ CREATE FUNCTION ... $cmd$);
+
+  -- List the work_mem setting of each worker database
+  SELECT run_command_on_workers($cmd$ SHOW work_mem; $cmd$);
+
+.. note::
+
+  The :code:`run_command_on_workers` function and other manual propagation commands in this section can run only queries which return a single column and single row.
+
+Running on all Shards
+---------------------
+
+Each distributed table :code:`foo` has shards across the workers which are each tables with a naming convention of :code:`foo_n` where :code:`n` is a number. In order to run a query against a shard, the query must be run on its containing worker, and the shard name must be interpolated into the query string as an identifier (:code:`%I`).
```

Comment:
Can each opening paragraph follow a similar structure? The opening paragraph for `run_command_on_workers` is nice, as it talks about what the command does, and why its useful.

This paragraph deviates a bit from that as it first talks about what shards are and their naming. It gives more of a 'how', rather than 'what' and 'why'. A more simple description could be something like: "This functions takes the given command and runs it across all shards for the given distributed table. It could be useful for ...". We don't need to dive into a discussion of shard name extension.